specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,280 @@
+- vertical-threshold - Numeric value that defines how small the region must be to the edge of the viewport to switch to the opposite side of the anchor. The component favors the default position until this value is crossed.  When there is not enough space on either side or the value is unset the side with the most space is chosen.
+- vertical-scaling-enabled - The region is sized from code to match available space, in other scenarios the region gets sized via content size.
+
+- horizontal-position - read only, the current horizontal position of the component. Possible values are 'left', 'right' or 'unset'.
+- vertical-position - read only, the current vertical position of the component. Possible values are 'top', 'bottom' or 'unset'.
+
+*Slots:*
+- default slot for content
+
+*Events:*
+- change - event is thrown whenever the placement of the region relative to the anchor changes (top/bottom and left/right). 
+
+
+### Anatomy and Appearance
+**Structure:**
+```
+<div class="root">
+    <slot></slot>
+</div>
+```
+
+**Appearance:**
+
+Parts:
+- root - the root of the anchored-region
+
+---
+
+## Implementation
+
+### States
+Layout update checks in the component happen when:
+- the viewport scrolls
+- intersection observer reports a collision with the viewport
+- resize observer reports a resize event on the anchor, the viewport or the component itself.
+
+These layout checks analyse the DOM geometry based on callbacks and repositions the anchored region appropriately: top/bottom/unset for the vertical axis and left/right/unset for the horizontal axis. 
+
+The component will have an initialization state to determine placement where the instanciation of content will be delayed by one frame to give the component a chance to get placed correctly.  This is to avoid content triggering the browser's scroll into view behavior prematurely if a contained element gains focus immediately.
+
+### Accessibility
+None required.  Basically a positioned div that authors can decorate for accessibility if required.
+
+### Globalization
+Authors may want to change default position from left to right or vice versa based on rtl settings, but that can't be predicted by the component itself.
+
+### Performance
+Layout logic could called frequently during scrolling and resize operations so efficient code is a priority here. At a minimum layout recalculations should be limited to once per frame.
+
+### Dependencies
+[IntersectionObserver api](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) is unsupported on IE, and [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) is unsupported on IE, Safari and Firefox.  Both are required by the component.
+
+Question - do we support a fallback mode that relies on getBoundingClientRect() if the observers are not present? 
```

Comment:
Maybe hold off on this for now. I think there are polyfills for the above, so maybe we can leverage them instead.

Reply:
Yup